test/src/math/test_vecmat.cpp

```diff
@@ -0,0 +1,353 @@
+#include "math/staticrand.h"
+#include "math/vecmat.h"
+#include "util/FSTestFixture.h"
+
+#include <gtest/gtest.h>
+
+class VecmatTest : public test::FSTestFixture {
+  public:
+	VecmatTest() : test::FSTestFixture() { pushModDir("vecmat"); }
+
+  protected:
+	void SetUp() override { test::FSTestFixture::SetUp(); }
+	void TearDown() override { test::FSTestFixture::TearDown(); }
+};
+
+TEST_F(VecmatTest, test_vm_vec_add)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v3;
+
+		static_randvec(rand32(), &v1);
+		static_randvec(rand32(), &v2);
```

Comment:
The premium solution here would be to use rapidcheck (or another quickcheck clone)
https://github.com/emil-e/rapidcheck

Not necessary, but really cool